### PR TITLE
Fix some NVCC warnings 

### PR DESCRIPTION
**Pull request by cyyever**
PR #95568 enables more NVCC warnings. However, some cu files need to be modified to make building process more warning free. Therefore, this work aims to fix them.

### CODE REVIEW

File: aten/src/ATen/native/cuda/CUDAJitLoops.cuh [hunk 1]
```diff
@@ -156,6 +156,8 @@ void launch_jitted_vectorized_kernel(
     at::cuda::jit::launch_jitted_pwise_function(
         *fn_ptr, args.data(), {grid, 1u, 1u}, {num_threads(), 1u, 1u});
   } else {
+#pragma nv_diagnostic push
+#pragma nv_diag_suppress 177
```
Comment (Skylion007) [inline aten/src/ATen/native/cuda/CUDAJitLoops.cuh:160]: Add a comment about what warning we are suppressing here and why?

Comment (cyyever) [reply aten/src/ATen/native/cuda/CUDAJitLoops.cuh:162]: Added in commit 0cadec0

File: aten/src/ATen/native/cuda/Loss.cu [hunk 2]
```diff
@@ -159,6 +161,13 @@ constexpr int NLL_LOSS_THREADS = 32;
   AT_PRIVATE_CASE_TYPE_USING_HINT(at::ScalarType::Byte, index_t, __VA_ARGS__) \
   AT_PRIVATE_CASE_TYPE_USING_HINT(at::ScalarType::Long, index_t, __VA_ARGS__))
 
+#define CHECK_INDEX_IN_CLASS(INDEX, N_CLASSES)                                \
+  if constexpr(std::is_unsigned<decltype(INDEX)>::value) {                    \
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/Loss.cu:165]: oh wow constexpr! cc @malfet

File: aten/src/ATen/native/cuda/SegmentReduce.cu [hunk 3]
```diff
@@ -62,24 +62,6 @@ struct CustomMin {
   }
 };
 
-Tensor _get_complete_sum(const Tensor& lengths) {
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/SegmentReduce.cu:65]: weird, why are we not using this?

**Review by ngimel — APPROVED** [review]

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**